mapomatic/layouts.py

```diff
@@ -39,13 +39,18 @@ def matching_layouts(circ, cmap, strict_direction=False, call_limit=10000):
 
     Parameters:
         circ (QuantumCircuit): Input quantum circuit
-        cmap (list): Coupling map
+        cmap (list or IBMQBackend): Coupling map or backend instance
         strict_direction (bool): Use directed coupling
         call_limit (int): Max number of calls to VF2 mapper
 
     Returns:
         list: Found mappings.
     """
+    if isinstance(cmap, list):
+        cmap = CouplingMap(cmap)
+    else:
+        cmap = CouplingMap(cmap.configuration().coupling_map)
```

Comment:
This isn't compatible with `BackendV2` which is fine, but moving forward more things will start to adopt the use of `BackendV2` so maybe add:
```python
elif isinstance(backend, BackendV2):
    cmap = backend.coupling_map
```
before this.

Reply:
Yeah I have not played with V2 yet, but would be good to have as well